inspirehep/utils/formulas.py

```diff
@@ -0,0 +1,180 @@
+        HTMLParser.__init__(self)
+        self.reset()
+        self.fed = []
+
+    def handle_data(self, d):
+        """Return representation of pure text data."""
+        self.fed.append(d.strip())
+
+    def handle_entityref(self, name):
+        """Return representation of entities."""
+        self.fed.append('&%s;' % name)
+
+    def handle_charref(self, name):
+        """Return representation of numeric entities."""
+        self.fed.append('&#%s;' % name)
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'msqrt':
+            self.fed.append(u'&radic;(')
+
+    def handle_endtag(self, tag):
+        if tag == 'msqrt':
+            self.fed.append(u')')
+
+    def get_data(self):
+        """Return all the stripped data."""
+        return self.unescape(''.join(self.fed))
+
+
+def splitter(formula):
+    """Split a formula.
+
+    Given a unicode-based formulas, returns the set of its component, namely
+    * the formula itself,
+    * split after and before the equal sign,
+    * split after and before any arrow,
+    * consider inner content of parenthesis.
+    """
+    if not valid_component(formula):
+        return []
+    if u'=' in formula:
+        return [formula] + [splitter(component) for component in formula.split(u'=')]
+    if u'→' in formula:
+        return [formula] + [splitter(component) for component in formula.split(u'→')]
+    if u'/' in formula:
+        return [formula] + [splitter(component) for component in formula.split(u'/')]
+    return [formula]
+
+
+def parenthesizer(formula):
+    """Parenthesised components of a formula.
+
+    Returns the parenthesized components of a formula.
+    E.g.
+        >>> parenthesizer("((a+b)*(c+d))/a")
+        ["((a+b)*(c+d))/a", "a+b", "c+d", "(a+b)*(c+d)"]
+    """
+    ret = [formula]
+    components = []
+    for i, c in enumerate(formula):
+        if c == u'(':
+            components.append(i)
+        elif c == u')':
+            start = components.pop()
+            ret.append(formula[start+1:i])
+    return ret
+
+
+def flatten(elements):
+    """Transforms a list of list into a flat list, recursively."""
+    ret = []
+    for element in elements:
+        if isinstance(element, list):
+            ret += flatten(element)
+        else:
+            ret.append(element)
+    return ret
+
+
+def formula2components(formula):
+    """Return the set of components of a unicode formula."""
+    return set(flatten(splitter(component) for component in parenthesizer(formula)))
+
+
+def valid_component(component):
+    """Check if a component has valid parentheses."""
+    count = 0
+    for c in component:
+        if c == '(':
+            count += 1
+        elif c == ')':
+            count -= 1
+        if count < 0:
+            return False
+    return count == 0
+
+
+def tex2mml(formula):
+    """Converts a LaTeX formula to MathML."""
+    try:
+        mathjax_node_server = current_app.config['MATHJAX_NODE_SERVER']
```

Comment:
The issue was to protect against working outside of the context.
`current_app.config.get('MATHJAX_NODE_SERVER', MATHJAX_NODE_SERVER)` would still raise `RuntimeError` outside of the context.